### PR TITLE
Helper script for downstream analysis

### PR DESCRIPTION
**Pull request by szilvajuhos**
This script (and docs) helps to add RankScore annotation to VCFs, so we can load them into Scout. 

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [x] Scout tests looks OK
 - [x] Ensure the test suite passes (`nextflow run . -profile test,docker`). `nextflow run . -profile test,singularity` is working, dunno what is wrong with docker (left for @MaxUlysse )
 - [x] Make sure your code lints (`nf-core lint .`). - lint needs some config variables also
 - [x] Documentation in `docs` is updated
 - [x] `CHANGELOG.md` is updated
 - [X] `README.md` is updated

**Learn more about contributing:** [guidelines](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: docs/downstream.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+# Downstream processing, ranking and connection to [Scout](https://github.com/Clinical-Genomics/scout)
+
+## RankScore annotations
+
+[Scout](https://github.com/Clinical-Genomics/scout) was developed to help clinicians to visualise relevant
+variations, focusing on rare diseases. However, adding cancer capabilities are on the way, and the first step to add
+somatic mutations is to rank them in the VCF and load them to Scout. The scores have to be added as a `RankScore` INFO
+field, and have to have the format like:
+
+```
```
Comment (MaxUlysse) [inline docs/downstream.md]:
```suggestion
```text
```

File: docs/downstream.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+chr1  514206  rs1247069502  A C . PASS  ECNT=1;dbSNP=rs1247069502 RankScore=ScoreTest:0 GT:AD:AF  0/1:2,6:0.750 0/0:33,1:0.029
+```
+
+where the `ScoreTest` is the sample (normal) ID, and the score of this particular variation is 0. Scout will accept and
+show any sort of scores if the VCF is annotated properly. There have to be an info line in the VCF like:
+
+```
```
Comment (MaxUlysse) [inline docs/downstream.md]:
```suggestion
```text
```

File: docs/downstream.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+##INFO=<ID=RankScore,Number=.,Type=String,Description="The rank score for this variant ... . family_id:rank_score.">
+``` 
+
+RankScores for Scout considering rare diseases are calculated by [genmod](https://github.com/moonso/genmod), hence we
+are following the same annotation format to accomodate VCF files generated by Sarek.
+
+## Calculating RankScores with [Pathfindr](https://github.com/NBISweden/pathfindr)
```
Comment (MaxUlysse) [inline docs/downstream.md]:
```suggestion
## Calculating RankScores with [Pathfindr](https://github.com/NBISweden/pathfindr)
```

File: docs/downstream.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+## Calculating RankScores with [Pathfindr](https://github.com/NBISweden/pathfindr) 
+
+There are many approaches to score variants, Pathfindr considers many statistics found in the VCF file, collates
+[snpEff](http://snpeff.sourceforge.net/) and [VEP](https://www.ensembl.org/info/docs/tools/vep/index.html) annotations
+to calculate a single score that is written into a HTML page, also into a CSV file. This CSV can be processed with the
+`scripts/addPFRanksToVCF.py` python3 script. The CSV do not have to be generated by Pathfindr: until you have a
+comma-delimited file with a header with the `ID` in the first column and further columns named as `chr`,`start` and
+`rank_score`, the VCF will be annotated with the provided rank score in the CSV:
+
+```
```
Comment (MaxUlysse) [inline docs/downstream.md]:
```suggestion
```text
```

File: docs/downstream.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+rs1335878992,SampleTheBig,2,T2_gene,"",0.00478691,MODIFIER,0,0.66847,chr2,80397367,80397367,1
+rs56269417,SampleTheBig,2,T2_gene,"",0.76779,MODIFIER,0,0.378918,chr2,98265733,98265733,1
+...
+```
+
+The basis of the annotation is `chr:startposition`: all variants in the VCF with the same coordinates will get the same
+RankScore irrespectively of the ALT allele. If there are multiple scores for a single genomic coordinate in the CSV, the
+very last one in the file will be stored only.
+
+## Usage of `scripts/addPFRanksToVCF.py` : transferring Pathfindr rank_score to Scout:
```
Comment (MaxUlysse) [inline docs/downstream.md]:
```suggestion
## Usage of `scripts/addPFRanksToVCF.py` : transferring Pathfindr rank_score to Scout
```

File: docs/downstream.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+Usage is:
+```
```
Comment (MaxUlysse) [inline docs/downstream.md]:
```suggestion

```text
```

File: docs/downstream.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+Usage: addPFRanksToVCF.py [OPTIONS]
+
+Options:
+  -v, --vcf TEXT     VCF file to annotate  [required]
+  -c, --csv TEXT     CSV file with ranked scores  [required]
+  -f, --family TEXT  family ID as expected in scout yaml file (usually sample
+                     name of the normal)  [required]
+  -h, --help         Show this message and exit.
+```
+Three parameters have to be provided, `-v` and `-c` are the name of the VCF and CSV files respectively, the `-f`
```
Comment (MaxUlysse) [inline docs/downstream.md]:
```suggestion

Three parameters have to be provided, `-v` and `-c` are the name of the VCF and CSV files respectively, the `-f`
```

File: docs/downstream.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+parameter is the name of the normal sample in the VCF file. MuTect2 (GATK version < 4.0) VCF has a header line like:
+
+```
```
Comment (MaxUlysse) [inline docs/downstream.md]:
```suggestion
```text
```

File: docs/downstream.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  TUMOR   NORMAL
+```
+
+not showing sample names, only the `TUMOR` and `NORMAL` columns. For Scout, the `NORMAL` part has to be altered to
+reflect actual sample name: this sample name will be also used in the RankScore annotation. By providing the something
+like `-f ThisIsIt` , the header line will be 
+
+```
```
Comment (MaxUlysse) [inline docs/downstream.md]:
```suggestion
```text
```

File: docs/downstream.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  TUMOR  ThisIsIt 
+```
+
+and annotations will be like `RankScore=ThisIsIt:7` . The `.yaml` file for Scout to load the case has to be prepared in
+a way, that the `family` entry and the `sample_id` in the `samples:` sections also contain this key. Example:
+
+```
```
Comment (MaxUlysse) [inline docs/downstream.md]:
```suggestion
```text
```

File: docs/downstream.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+Three parameters have to be provided, `-v` and `-c` are the name of the VCF and CSV files respectively, the `-f`
+parameter is the name of the normal sample in the VCF file. MuTect2 (GATK version < 4.0) VCF has a header line like:
+
+```
+#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  TUMOR   NORMAL
+```
+
+not showing sample names, only the `TUMOR` and `NORMAL` columns. For Scout, the `NORMAL` part has to be altered to
+reflect actual sample name: this sample name will be also used in the RankScore annotation. By providing the something
+like `-f ThisIsIt` , the header line will be
```
Comment (MaxUlysse) [inline docs/downstream.md]:
```suggestion
like `-f ThisIsIt` , the header line will be
```

File: docs/downstream.md [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+```
+#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  TUMOR   NORMAL
+```
+
+not showing sample names, only the `TUMOR` and `NORMAL` columns. For Scout, the `NORMAL` part has to be altered to
+reflect actual sample name: this sample name will be also used in the RankScore annotation. By providing the something
+like `-f ThisIsIt` , the header line will be 
+
+```
+#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  TUMOR  ThisIsIt
```
Comment (MaxUlysse) [inline docs/downstream.md]:
```suggestion
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  TUMOR  ThisIsIt
```

**Comment by MaxUlysse** [comment]
Looks good, just need to fix the Markdown linting ;-)

**Comment by MaxUlysse** [comment]
@szilvajuhos any updates on this one?

File: docs/downstream.md [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+Options:
+  -v, --vcf TEXT     VCF file to annotate  [required]
+  -c, --csv TEXT     CSV file with ranked scores  [required]
+  -f, --family TEXT  family ID as expected in scout yaml file (usually sample
+                     name of the normal)  [required]
+  -h, --help         Show this message and exit.
+```
+
+Three parameters have to be provided, `-v` and `-c` are the name of the VCF and CSV files respectively, the `-f`
+parameter is the name of the normal sample in the VCF file. MuTect2 (GATK version < 4.0) VCF has a header line like:
```
Comment (MaxUlysse) [inline docs/downstream.md]:
```suggestion
parameter is the name of the normal sample in the VCF file. Mutect2 (GATK version < 4.0) VCF has a header line like:
```

Comment (MaxUlysse) [reply docs/downstream.md]: Do we still use this version?

Comment (szilvajuhos) [reply docs/downstream.md]: - for GATK 3.8 mutect is called MuTect2
 - for GATK 4.x mutect is called Mutect2

We do not use GATK 3.8, but I we still have data already analysed by old MuTect2.

**Comment by MaxUlysse** [comment]
So what's your plan with this PR now?

**Comment by szilvajuhos** [comment]
Moved already to btb-scripts: https://github.com/szilvajuhos/btb-scripts/commit/89f4a662d5b96ea2149bd0acbbcc463e60d7d1bf . The only thing in the diff we can consider to keep is the CPU requirement for pilepu merge https://github.com/nf-core/sarek/pull/51/files#diff-28b7bea57b04f15beeb4c1de0ee78382R2910

**Comment by MaxUlysse** [comment]
I see, do you want to keep that, I'm making a PR now to add some tests, I can easily squeeze that in

**Comment by szilvajuhos** [comment]
Not at all, we can keep it independent of Sarek, get rid of it.

**Comment by MaxUlysse** [comment]
I meant squeezing in the 1 cpu requirement for the pileupmerge process

**Comment by szilvajuhos** [comment]
Oh, that one, yes please, squeeze in.

**Pull request closed without merging**